hypervisor/arch/x86/trusty.c

```diff
@@ -417,48 +417,38 @@ static bool init_secure_world_env(struct vcpu *vcpu,
 bool initialize_trusty(struct vcpu *vcpu, uint64_t param)
 {
 	uint64_t trusty_entry_gpa, trusty_base_gpa, trusty_base_hpa;
+	uint64_t trusty_mem_size;
 	struct vm *vm = vcpu->vm;
 	struct trusty_boot_param boot_param;
 
 	memset(&boot_param, 0, sizeof(boot_param));
-	if (copy_from_gpa(vcpu->vm, &boot_param, param, sizeof(boot_param))) {
+	if (copy_from_gpa(vcpu->vm, &boot_param, param,
+					sizeof(boot_param))) {
 		pr_err("%s: Unable to copy trusty_boot_param\n", __func__);
 		return false;
 	}
 
-	/* FIXME: Temporarily comments out the size check since need to extend
-	 *        the interface, need to add back after the integration done.
-	 */
-	/*
-	if (sizeof(struct trusty_boot_param) !=
-			boot_param.size_of_this_struct) {
-		pr_err("%s: sizeof(struct trusty_boot_param) mismatch!\n",
-			__func__);
-		return false;
-	}
-	*/
+	if (boot_param.version == TRUSTY_VERSION) {
```

Comment:
how about just always assign the lower 32bit without checking version?
then check when the version is 2, will append high 32bit.

Reply:
Done. Please check latest code.